main.go

```diff
@@ -85,38 +85,63 @@ func main() {
 		log.Fatalf(usage)
 	}
 
-	url, err := url.Parse(args[0])
+	url := parseURL(args[0])
+
+	visit(url)
+}
+
+func parseURL(uri string) *url.URL {
+	url, err := url.Parse(schemify(uri))
 	if err != nil {
-		log.Fatalf("could not parse url %q: %v", args[0], err)
+		log.Fatalf("could not parse url %q: %v", uri, err)
 	}
+	return url
+}
 
-	visit(url)
+func schemify(uri string) string {
+	if strings.Contains(uri, "://") != true {
```

Comment:
Maybe it's just me, but I find the long-form of this expression unexpected. Any objections to just evaluating the return of the function itself:

```Go
if !strings.Contains(uri, "://") { ... }
```